server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java

```diff
@@ -107,8 +108,58 @@
 import com.beust.jcommander.Parameter;
 import com.google.common.base.Preconditions;
 
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+
 public class Compactor extends AbstractServer implements CompactorService.Iface {
 
+  private static class CompactionProgress implements MetricsProducer {
+    private final long warnTime;
+    private long time;
+    private long read;
+    private long written;
+    private boolean stuck = false;
+
+    public CompactionProgress(AccumuloConfiguration config) {
+      warnTime = config.getTimeInMillis(Property.TSERV_COMPACTION_WARN_TIME);
+    }
+
+    @Override
+    public void registerMetrics(MeterRegistry registry) {
+      Gauge.builder(METRICS_TSERVER_MAJC_STUCK, this, CompactionProgress::getNumStuck)
+          .description("Number of stuck major compactions").register(registry);
+    }
+
+    public long getNumStuck() {
+      return stuck ? 1 : 0;
+    }
```

Comment:
Something like if not stuck return 0, else return System.currentTimeMillis() - time